web-app/src/actions/dashboard.js

```diff
@@ -63,30 +64,37 @@ export const setEventView = eventView => {
 
 // Thunks
 
-export const fetchEvents = (date, eventView) => dispatch => {
-  dispatch(setLoading(true));
-
-  if (eventView === eventsView.PERSONAL_EVENTS) {
-    getUsersEvents(date)
-      .then(({ data }) => {
-        dispatch(setLoading(false));
-        const formattedEvents = transformEvents(data);
-        dispatch(setCalendarEvents(formattedEvents));
-      })
-      .catch(error => {
-        dispatch(setLoading(false));
-        dispatch(setError(error));
-      });
-  } else if (eventView === eventsView.TEAM_EVENTS) {
-    getTeamsEvents(date)
-      .then(({ data }) => {
-        dispatch(setLoading(false));
-        const formattedEvents = transformEvents(data.events);
-        dispatch(setCalendarEvents(formattedEvents));
-      })
-      .catch(error => {
-        dispatch(setLoading(false));
-        dispatch(setError(error));
-      });
+export const fetchEvents = (date, eventView, force = false) => dispatch => {
+  // Check if we need to make a request (is there existing state?)
+  if (requiresNewRequest(date) || force) {
+    // Start loading
+    dispatch(setLoading(true));
+    // Fetch personal events only.
+    if (eventView === eventsView.PERSONAL_EVENTS) {
+      getUsersEvents(date)
+        .then(({ data }) => {
+          dispatch(setLoading(false));
+          transformEvents(data).then(transformedEvents => {
+            dispatch(setCalendarEvents(transformedEvents));
+          });
+        })
+        .catch(error => {
+          dispatch(setLoading(false));
+          dispatch(setError(error));
+        });
+      // Fetch team's events as well as your own.
+    } else if (eventView === eventsView.TEAM_EVENTS) {
+      getTeamsEvents(date)
+        .then(({ data }) => {
+          dispatch(setLoading(false));
```

Comment:
The fact the code in the block on line 89 onwards plus line 76 onwards is the same....maybe we could pull this out and reuse it within a function?

Reply:
Make sense. Will do. 👍